src/topology/config/mod.rs

```diff
@@ -160,6 +162,25 @@ pub trait TransformConfig: core::fmt::Debug {
     fn transform_type(&self) -> &'static str;
 }
 
+pub struct BoxTransformConfig {
+    pub inner: Box<dyn TransformConfig>,
+}
```

Comment:
I'm not totally clear why we need this type.

Reply:
This is a little bit of a headache but for inventory we need a sizeable type, so we need `Box<dyn Foo>`. We also want to be able to register the plugin config type itself rather than a boxed trait as that allows us to access `Serialize`, etc, in our `new` implementation.

However, rust can't infer the implementation of `Into` for a `T` into a boxed trait as it requires two jumps, the first being `T` into `dyn Foo`, the second being `dyn Foo` into `Box<dyn Foo>`, that means we can't just simply register `FooDescription::new::<BarConfig>("bar")`. But we could probably settle for something like `FooDescription::new::<Box<dyn Foo>, BarConfig>("bar")`.

I was aiming to shorten the per-plugin boilerplate, which we can do by defining  and registering `BoxTransformConfig` and implementing a generic `BoxTransformConfig::From<T>` which covers three steps, that allows us to use just `FooDescription::new::<BarConfig>("bar")`.

Although thinking about it, if it's possible to alias a function we can probably do something like:

`type RegisterFoo<T> = FooDescription::new::<Box<dyn Foo>, T>;`

Followed by lots of `RegisterFoo::<BarConfig>("bar")`. Can't check right now but if it's possible then I prefer it.

Reply:
Actually scratch that, we aren't using `T` into `Box<dyn Foo>` anymore because we aren't deserializing into it. I can get rid of the `Into<T>` trait bounds and remove `BoxTransformConfig`. I'll update the PR tomorrow morning.